pkg/kv/kvserver/replica_test.go

```diff
@@ -854,6 +854,20 @@ func TestLeaseReplicaNotInDesc(t *testing.T) {
 	}
 }
 
+// TestReplicaRangeBoundsChecking tests two behaviors that should occur.
+// (1): Following a Range split, however prior to the RHS Range being
```

Comment:
This is really helpful detail! And a good test for what we've added. I'd suggest renaming it a bit more appropriately given what it now does. Also [nit] there's something off with the first two sentences here. How about the following?

```
- Following a Range split, the client may send BatchRequests based on stale
  cache data targeting the wrong range. Internally this triggers a
  RangeKeyMismatchError, but in the cases where the RHS of the range is still
  present on the local store, we opportunistically retry server-side by
  re-routing the request to the right range. No error is bubbled up to the
  client.
- This test also ensures that after a successful server-side retry attempt we
  bubble up the most up-to-date RangeInfos for the client to update its range
  cache.
```